brushmodes.c

```diff
@@ -209,10 +209,10 @@ void draw_dab_pixels_BlendMode_Normal_and_Eraser (uint16_t * mask,
 
   while (1) {
     for (; mask[0]; mask++, rgba+=4) {
-      uint32_t opa_a = mask[0]*(uint32_t)opacity/(1<<15); // topAlpha
+      uint32_t opa_a = (mask[0]*(uint32_t)opacity) >> 15; // topAlpha
       uint32_t opa_b = (1<<15)-opa_a; // bottomAlpha
-      opa_a = opa_a * color_a / (1<<15);
-      rgba[3] = opa_a + opa_b * rgba[3] / (1<<15);
+      opa_a = (opa_a * color_a) >> 15;
+      rgba[3] = (opa_a + opa_b * rgba[3]) >> 15;
```

Comment:
:-1: Precedence issues. Multiplication/division binds tighter than addition, and both bind tighter than bitshifting. Suggestion: bracket up the 1st `rgba[3]` assignment, then convert to bitshifts.